### DISCUSSION TITLE
Removing keymap for <leader>qq

### DISCUSSION THREAD

**Discussion by varunshoor** [LazyVim/LazyVim#90] [Q&A] [answered]
Hi,
How do I remove the keymap entry for `<leader>qq`? I am using to using `<leader>q` for deleting the buffer and the `<leader>qq` entry has created a delay.

I tried:
`vim.keymap.set("n", "<leader>qq", "<Nop>")`

The mapping gets removed but the lag remains.

Please help. I have checked all around and Google isn't of much help.

**Comment by loiccoyle** [comment]
Did you try to delete the keymap with `vim.keymap.del("n", "<leader>qq")`? Check the help page with `:help vim.keymap.del()`.

**Comment by varunshoor** [comment, marked as answer]
Thanks so much for responding! I did do that but surprisingly the delay still remains. That command should clear the keymap, right?

I also tried adding `nowait = true` but that didn't solve it either

```
keymap.del("n", "<leader>qq")
keymap.set("n", "<leader>q", function()
  require("mini.bufremove").delete(0, false)
end, { nowait = true }) 
```

I was wondering if it was the `mini.bufremove` call that would be causing the delay so I replaced the key with `<leader>z` and that did remove the delay. So it has to be something where vim is continuing to keep the `<leader>qq` in register somehow?

**Reply by loiccoyle** [reply]
Did you  also remove the other keymaps also bound which start with the `<leader>q` chord chain, like the session management keys?
 https://github.com/LazyVim/LazyVim/blob/6a3f3d3032181dae5fab36b5acc9ea5da086e7fa/lua/lazyvim/plugins/util.lua#L12

**Reply by folke** [reply]
As @loiccoyle mentioned, you need to disable all the mappings under `<leader>q`

**Reply by varunshoor** [reply]
This might be it! Will check and circle back tomorrow. Thank you so so much!

**Comment by pchabros** [comment]
Hi,

I tried to remove mappings for `K` and `S` with `vim.keymap.del`:

```
vim.keymap.del("n", "K")
vim.keymap.del("n", "S")
```

and I get an error:

```
E31: no such mapping
```

Please help - lack of `S` slows me down.

**Reply by folke** [reply]
https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps

And disable leap, or change its config

**Reply by pawelchabros** [reply]
I've added `lua/plugins/leap.lua`:

```
return {
  "ggandor/leap.nvim",
  config = function()
    require("leap").add_default_mappings(false)
  end,
}
```

And it has no effect.

**Reply by pawelchabros** [reply]
This works:

```
require("leap").add_default_mappings = false
```

Do you know why it works while this don't:

```
require("leap").add_default_mappings(false)
```

I don't know lua, but `add_default_mappings` looks like a setter, and overwriting function with `false` works?

**Reply by folke** [reply]
Just don't add that line and keep an empty config function

**Reply by pawelchabros** [reply]
Makes sense. Thanks!

**Comment by dnxxymcj** [comment]
--leap.add_default_mappings(true)
  
delete this line and custom keys

**Reply by pawelchabros** [reply]
I would like to change it from my `.config/nvim`.

**Reply by dnxxymcj** [reply]
```lua
  -- customize leap     leap.lua
  {
    "ggandor/leap.nvim",
    event = "VeryLazy",
    dependencies = {
      {
        "ggandor/flit.nvim",
        opts = { labeled_modes = "nv" },
        -- keys = { f = "f", F = "F" },
      },
    },
    config = function(_, opts)
      local leap = require("leap")
      for k, v in pairs(opts) do
        leap.opts[k] = v
      end
      leap.opts.safe_labels = {}
      leap.opts.labels = { "s", "f", "t", "r", "n", "e", "i", "o", "d", "w", "l", "m", "u", "y", "k", "v", "h", "g", "h", "c", "x", "/", "a", "p" }
      -- leap.add_default_mappings(true)
      vim.keymap.set({ "x", "o" }, "h", "<Plug>(leap-forward-till)", { noremap = true, silent = true })
      vim.keymap.set({ "x", "o" }, "H", "<Plug>(leap-backward-till)", { noremap = true, silent = true })
      vim.keymap.set("n", "h", "<Plug>(leap-forward-to)", { noremap = true, silent = true })
      vim.keymap.set("n", "H", "<Plug>(leap-backward-to)", { noremap = true, silent = true })

      vim.api.nvim_set_hl(0, "LeapBackdrop", { link = "Comment" })
      vim.api.nvim_set_hl(0, "LeapMatch", {
        fg = "white", -- for light themes, set to 'black' or similar
        bold = true,
        nocombine = true,
      })
      leap.opts.highlight_unlabeled_phase_one_targets = true
    end,
  },

```

**Reply by pawelchabros** [reply]
I don't want to copy and paste whole config. I just wanted to change `add_default_mappings`.

I manged to disable it with just:

```
return {
  "ggandor/leap.nvim",
  config = function()
    require("leap").add_default_mappings = false
  end,
}
```
